tests/test_conn.py

```diff
@@ -116,3 +116,125 @@ def test_delete_user_mailcow_missing(conn, mailcow):
 def test_db_version(conn):
     version = conn.get_dbversion()
     assert type(version) == int
+
+
+def test_users_to_warn(conn, monkeypatch):
+    yeartoken = conn.add_token("1y", token="asd1", expiry="1y", prefix="tmp.")
+    monthtoken = conn.add_token("1m", token="asd2", expiry="1m", prefix="tmp.")
+    weektoken = conn.add_token("1w", token="asd3", expiry="1w", prefix="tmp.")
+    daytoken = conn.add_token("1d", token="asd4", expiry="1d", prefix="tmp.")
+
+    yearuser = conn.add_email_account(yeartoken)
+    monthuser = conn.add_email_account(monthtoken)
+    weekuser = conn.add_email_account(weektoken)
+    dayuser = conn.add_email_account(daytoken)
```

Comment:
a bit skeptical of this test_users_to_warn functions -- a bit intricate to follow and a lot of boilerplate.  When we change the logic/ warning rhythm it might also break easily.  but not sure it's worth to refactor now.  The other point (of having a get_users_to_warn that produces messages fitting tot he current time) is probably more important.